clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -469,7 +470,7 @@ private void processDisconnection(List<ClientResponse> responses, String nodeId,
         nodeApiVersions.remove(nodeId);
         nodesNeedingApiVersionsFetch.remove(nodeId);
         for (InFlightRequest request : this.inFlightRequests.clearAll(nodeId)) {
-            log.trace("Cancelled request {} due to node {} being disconnected", request, nodeId);
+            log.trace("Cancelled request {} due to node {} being disconnected", request.requestBuilder, nodeId);
```

Comment:
Thinking about it a bit more, it seems struct's toString is the way to go.

It's harder to mess up by using the underlying request struct. Builder toStrings are updated manually while struct toString just traverses the underlying struct so it will automatically pick up wire format changes.